app/assets/javascripts/discourse/app/components/user-notification-schedule.js

```diff
@@ -0,0 +1,147 @@
+  model: null,
+
+  @action
+  onChangeStartTime(val) {
+    this.startingTimeChangedForDay(val);
+  },
+
+  @action
+  onChangeEndTime(val) {
+    this.set(`model.user_notification_schedule.day_${this.id}_end_time`, val);
+  },
+
+  @discourseComputed(
+    "model.user_notification_schedule.day_{0,1,2,3,4,5,6}_start_time"
+  )
+  startTimeValue(schedule) {
+    return schedule[`day_${this.id}_start_time`];
+  },
+
+  @discourseComputed(
+    "model.user_notification_schedule.day_{0,1,2,3,4,5,6}_start_time"
+  )
+  endTimeOptions(schedule) {
+    return this.buildEndTimeOptionsFor(schedule[`day_${this.id}_start_time`]);
+  },
+
+  @discourseComputed(
+    "model.user_notification_schedule.day_{0,1,2,3,4,5,6}_end_time"
+  )
+  endTimeValue(schedule) {
+    return schedule[`day_${this.id}_end_time`];
+  },
+
+  startingTimeChangedForDay(val) {
+    val = parseInt(val, 10);
+    this.model.set(`user_notification_schedule.day_${this.id}_start_time`, val);
+    if (
+      val !== "-1" &&
+      this.model.user_notification_schedule[`day_${this.id}_end_time`] <= val
+    ) {
+      this.model.set(
+        `user_notification_schedule.day_${this.id}_end_time`,
+        val + 30
+      );
+    }
+  },
+
+  buildEndTimeOptionsFor(startTime) {
+    startTime = parseInt(startTime, 10);
+    if (startTime === -1) {
+      return null;
+    }
+    return this.buildTimeOptions(startTime + 30, {
+      includeNone: false,
+      showMidnight: true,
+    });
+  },
+});
+
+export default Component.extend({
+  days: null,
+
+  didInsertElement() {
+    this._super(...arguments);
+    this.set(
+      "startTimeOptions",
+      this.buildTimeOptions(0, {
+        includeNone: true,
+        showMidnight: false,
+      })
+    );
+
+    this.set("days", []);
+    for (let i = 0; i < 7; i++) {
+      this.days.pushObject(
+        Day.create({
+          id: i,
+          day: daysMap[i],
+          model: this.model,
+          buildTimeOptions: this.buildTimeOptions,
+          startTimeOptions: this.startTimeOptions,
+        })
+      );
+    }
+  },
+
+  buildTimeOptions(startAt, opts = { includeNone: false, showMidnight: true }) {
+    let timeOptions = [];
+
+    if (opts.includeNone) {
+      timeOptions.push({
+        name: I18n.t("user.notification_schedule.none"),
+        value: -1,
+      });
+    }
+
+    for (let timeInMin = startAt; timeInMin <= 1440; timeInMin += 30) {
+      let hours = Math.floor(timeInMin / 60);
+      let minutes = timeInMin % 60;
+      let am = timeInMin < 720;
+      if (minutes === 0) {
+        minutes = "00";
+      }
+      if (hours === 0) {
+        hours = "12";
+      }
+      if (hours === 24) {
+        if (opts.showMidnight) {
+          timeOptions.push({
+            name: I18n.t("user.notification_schedule.midnight"),
+            value: 1440,
+          });
+        }
+        break;
+      }
+      if (!am && hours !== 12) {
+        hours = hours - 12;
+      }
+      const amPm = I18n.t(
```

Comment:
I see what you did here 😅 I'm really annoyed we are creating hardcoded locales with AM/PM, this is not how we show localised time.

Reply:
can you suggest something else for me?

Reply:
Maybe you create a fake moment with hours and minutes and just format it to show hours and minutes?

```
moment().set("hour", 12).set("minute", 28).format("LT")
// 12:28 PM (if moment.locale() === "en")
```

Reply:
@jjaffeux https://github.com/discourse/discourse/pull/11665/commits/9315a7e05e69f0b2992e93c9c73aeace13e9e93f